drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -653,6 +659,24 @@ static void nrf5_irq_config(const struct device *dev)
 #endif
 }
 
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
+void nrf_802154_tx_started(const uint8_t *aFrame)
+{
+       assert(aFrame == nrf5_data.tx_psdu);
+
+       if (nrf5_data.tx_psdu_time_ie_offset != 0) {
+               uint8_t *timeIe = (nrf5_data.tx_psdu + 1) + nrf5_data.tx_psdu_time_ie_offset;
+               uint64_t time = (uint64_t)((int64_t)nrf5_get_time(NULL) + nrf5_data.tx_network_time_offset);
+
+               *(++timeIe) = (uint8_t)(time & 0xff);
```

Comment:
Add a comment that the first byte was already populated and that's why it's skipped.